edweiss-app/app/(app)/quiz/createQuizPage.tsx

```diff
@@ -0,0 +1,612 @@
+			});
+			return false;
+		}
+
+		return true
+
+	}
+
+	async function publishQuiz() {
+
+		checkQuizParams()
+		if (dueDate == undefined) { // should not happen with check
+			return;
+		}
+		const answers = exercises.map(exercise => {
+			if (exercise.type == "MCQ") {
+				const answer: QuizzesAttempts.MCQAnswersIndices = { type: "MCQAnswersIndices", value: exercise.answersIndices }
+				return answer;
+			}
+			else { // if exercise is TF
+				const answer: QuizzesAttempts.TFAnswer = { type: "TFAnswer", value: exercise.answer }
+				return answer;
+			}
+		})
+		const newQuiz: Quizzes.Quiz = {
+			exercises: exercises,
+			dueDate: Time.fromDate(dueDate),
+			name: quizName,
+			ended: false,
+			showResultToStudents: false,
+			type: "quiz",
+			answers: answers
+
+		}
+
+		const res = await callFunction(Quizzes.Functions.createQuiz, { quiz: newQuiz, courseId: courseId });
+
+		if (res.status === 1) {
+			console.log(`OKAY, submitted quiz with id ${res.data.id}`);
+		} else {
+			console.log(`Error while submitting attempt`);
+		}
+	}
+
+	// 				<TText mb='md' ml='md' size={'xl'}> {quizName}</TText>
+
+	if (exercises.length <= 0) {
+		return (
+			<>
+				<NameAndDateCustom dueDate={dueDate} onChangeDate={onChangeDate} onChangeTime={onChangeTime} setQuizName={setQuizName} setShowDate={setShowDate} setShowTime={setShowTime} showDate={showDate} showTime={showTime}></NameAndDateCustom>
+
+				<TText ml='xl' mb='xl'>
+					Quiz does not have any exercise yet.
+				</TText>
+
+				<FancyButton icon='add' backgroundColor='blue' onPress={() => addModalRef.current?.present()} ></FancyButton>
+				<AddExerciseModal modalRef={addModalRef} updateExerciseList={addToExerciseList}></AddExerciseModal>
+			</>
+
+		)
+	}
+	return (
+		<>
+			<NameAndDateCustom dueDate={dueDate} onChangeDate={onChangeDate} onChangeTime={onChangeTime} setQuizName={setQuizName} setShowDate={setShowDate} setShowTime={setShowTime} showDate={showDate} showTime={showTime}></NameAndDateCustom>
+
+			<TScrollView>
+
+				<For each={exercises}>{
+					(exercise, index) => {
+						return (<>
+							<PressableExercise index={index} editExercise={editExercise} removeExerciseFromList={removeExerciseFromList} exercise={exercise} key={index}></PressableExercise>
+						</>)
+					}
+				}
+				</For>
+
+			</TScrollView>
+			<FancyButton icon='add' backgroundColor='blue' onPress={() => addModalRef.current?.present()} ></FancyButton>
+			<FancyButton onPress={() => { checkQuizParams() ? publishQuizModalRef.current?.present() : {} }}> Publish quiz </FancyButton>
+			<AddExerciseModal modalRef={addModalRef} updateExerciseList={addToExerciseList} />
+			<PublishQuizModal modalRef={publishQuizModalRef} quizName={quizName} dueDate={dueDate == undefined ? "undefined" : dueDate.toDateString()} numberOfExercises={exercises.length} publishQuiz={publishQuiz} />
+
+		</>
+
+	);
+};
+export default CreateQuizPage;
+
+function timestampToDate(timestamp: number): Date {
```

Comment:
Maybe you can add this function on utils file

Reply:
I will think about that for the next sprint!